### PR TITLE
feat(ingest) Athena: Getting table properties for Athena datasets

### PR DESCRIPTION
**Pull request by treff7es**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/sql/athena.py [hunk 1]
```diff
@@ -33,10 +37,57 @@ def get_sql_alchemy_url(self):
 
 
 class AthenaSource(SQLAlchemySource):
+    cursor: Optional[BaseCursor] = None
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/ingestion/source/sql/athena.py:40]: Should this be at the class-scope or be an instance member?

File: metadata-ingestion/src/datahub/ingestion/source/sql/athena.py [hunk 2]
```diff
@@ -33,10 +37,57 @@ def get_sql_alchemy_url(self):
...
+            table_name=table, schema_name=schema
+        )
+        description = metadata.comment
+        parameters = {k: v for k, v in metadata.parameters.items() if v}
+
+        custom_properties = {
+            **parameters,
+            **{
+                k: str(v)
+                for k, v in metadata.__dict__.items()
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/ingestion/source/sql/athena.py:69]: What exactly do we need here? Accessing `__dict__` seems quite risky here. Pretty much every private member seems to be accessible via the @property decorator.

File: metadata-ingestion/src/datahub/ingestion/source/sql/athena.py [hunk 3]
```diff
@@ -33,10 +37,57 @@ def get_sql_alchemy_url(self):
...
+                        "_parameters",
+                        "_name",
+                        "comment",
+                        "_partition_keys",
+                    ]
+                    and v
+                )
+            },
+        }
+        if metadata.partition_keys:
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/ingestion/source/sql/athena.py:83]: This is an unnecessary check. Default value is an empty array.

File: metadata-ingestion/src/datahub/ingestion/source/sql/athena.py [hunk 4]
```diff
@@ -33,10 +37,57 @@ def get_sql_alchemy_url(self):
...
+            partition_key = []
+            for key in metadata.partition_keys:
+                partition_key.append(key.__dict__)
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/ingestion/source/sql/athena.py:86]: What's going on here with key.__dict__? All of the members of `AthenaTableMetadataPartitionKey` are accessible via the @property decorator.

**Review by rslanka — CHANGES_REQUESTED** [review]
This needs an update. Please see comments for details.

File: metadata-ingestion/setup.py [hunk 5]
```diff
@@ -102,7 +102,7 @@ def get_long_description():
         "apache-airflow >= 1.10.2",
     },
     # Source plugins
-    "athena": sql_common | {"PyAthena[SQLAlchemy]"},
+    "athena": sql_common | {"PyAthena[SQLAlchemy]==2.4.1"},
```
Comment (rslanka) [inline metadata-ingestion/setup.py:105]: Why `==` and not `>=`? Would this be the same as `pyathena>=2.4.1`?

Comment (rslanka) [reply metadata-ingestion/setup.py:106]: Ah! I see...we are using private APIs!

**Review by rslanka — APPROVED** [review]
LGTM!

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**